drivers/serial.c

```diff
@@ -203,7 +218,6 @@ int serial_putchar(io_port_t port, char c) {
 
     do {
         rc = uart_putc(port, c);
-        BUG_ON(rc < 0);
     } while (rc > 0 && retries--);
```

Comment:
Maybe make the loop condition explicitly test for `EAGAIN`, or, even better, `-EAGAIN` and return that value in uart_port_status()?

Reply:
Ok